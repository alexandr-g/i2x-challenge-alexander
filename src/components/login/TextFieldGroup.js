import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';

const TextFieldGroup = ({ field, value, label, error, type, onChange, checkUserExists }) => {
  return (
    <div className={classnames('form-group', { 'has-error': error })}>
      <label className="db fw6 lh-copy f6">{label}</label>
      <input
        onChange={onChange}
        onBlur={checkUserExists}
        value={value}
        type={type}
        name={field}
        className="pa2 input-reset ba bg-transparent hover-bg-black hover-white w-100"
      />
      {error && <span className="help-block">{error}</span>}
    </div>  );
};

TextFieldGroup.propTypes = {
  field: PropTypes.string.isRequired,
  value: PropTypes.string.isRequired,
  label: PropTypes.string.isRequired,
  error: PropTypes.string,
  type: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  checkUserExists: PropTypes.func
};

TextFieldGroup.defaultProps = {
  type: 'text'
};

export default TextFieldGroup;
