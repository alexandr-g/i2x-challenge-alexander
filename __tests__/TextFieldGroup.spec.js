/* env jest */

import React from 'react';
import renderer from 'react-test-renderer';
import TextFieldGroup from '../src/components/login/TextFieldGroup';

describe('TextFieldGroup:', () => {
  test('is rendered', () => {
    const component = renderer.create(<TextFieldGroup field="email" label="E-mail" value="test@test.com" />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
