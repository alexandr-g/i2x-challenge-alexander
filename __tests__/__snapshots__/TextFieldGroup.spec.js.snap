exports[`TextFieldGroup: is rendered 1`] = `
<div
  className="form-group">
  <label
    className="db fw6 lh-copy f6">
    E-mail
  </label>
  <input
    className="pa2 input-reset ba bg-transparent hover-bg-black hover-white w-100"
    name="email"
    onBlur={undefined}
    onChange={undefined}
    type="text"
    value="test@test.com" />
</div>
`;
